### DISCUSSION TITLE
Scoop not working with Packer build

### DISCUSSION THREAD

**Discussion by ghost** [ScoopInstaller/Scoop#5537] [Help and Questions]
Hi, 

I have been trying to run scoop using Packer but when we test our image through a VM, the tools do not seem to have installed even though our pipeline shows that it ran successfully. 

I tried doing this interactively with my machine and it worked. 

Has anyone tried using scoop with packer by adding it in an image? or does scoop not work with Packer?

Any help would be highly appreciated.

**Comment by HUMORCE** [comment]
You should leave more information about this. e.g. console outputs, screenshots, configurations

**Reply by ghost** [reply]
**This is my ps1 file** 

`iex "& {$(irm get.scoop.sh)} -RunAsAdmin"

scoop update

scoop install git

scoop bucket add extras

scoop install rancher-desktop`

**This is the json file** 

{
  "variables": {
      "client_id": "{{env `ARM_CLIENT_ID`}}",
      "client_secret": "{{env `ARM_CLIENT_SECRET`}}",
      "subscription_id": "{{env `ARM_SUBSCRIPTION_ID`}}",
      "tenant_id": "{{env `ARM_TENANT_ID`}}",
      "build_version": "{{env `BUILD_BUILDNUMBER`}}.0"
  },
  "builders": [{
      "type": "azure-arm",
      "client_id": "{{user `client_id`}}",
      "client_secret": "{{user `client_secret`}}",
      "subscription_id": "{{user `subscription_id`}}",
      "tenant_id": "{{user `tenant_id`}}",
      
      "managed_image_resource_group_name": "hdsplt-devbox-dev-scus-rg",
      "managed_image_name": "hdsplt-devbox-dev-scus-wi",

      "build_resource_group_name": "hdsplt-devbox-dev-scus-rg",
      "virtual_network_name": "D-AZ-HDS-PLT-DEV-01-SVNET",
      "virtual_network_subnet_name": "mgmt",
      "virtual_network_resource_group_name": "D-AZ-HDS-PLT-DEV-01-SVRG",

      "os_type": "Windows",
      "image_publisher": "microsoftwindowsdesktop",
      "image_offer": "windows-ent-cpc",
      "image_sku": "win10-21h2-ent-cpc-os-g2",
      "vm_size": "Standard_D4ds_v5",
      
      "communicator": "winrm",
      "winrm_use_ssl": "true",
      "winrm_insecure": "true",
      "winrm_username": "packer",
      "winrm_port": "5986",

      "shared_image_gallery_destination": {
        "subscription": "{{user `subscription_id`}}",
        "resource_group": "hdsplt-devbox-dev-scus-rg",
        "gallery_name": "hdspltdevboxdevscuscg",
        "image_name": "hdsplt-devbox-dev-scus-wi",
        "image_version": "{{user `build_version`}}",
        "replication_regions": ["South Central US"]
      }
  }
],

"provisioners": 
[
  
  {
    "type": "powershell",
    "scripts": [
      "{{ template_dir }}/scripts/Install-Tools.ps1"
    ]
  },
  {
    "type": "windows-restart"
  },
  {
    "type": "powershell",
    "inline": [
      "while ((Get-Service RdAgent).Status -ne 'Running') { Start-Sleep -s 5 }",
      "while ((Get-Service WindowsAzureGuestAgent).Status -ne 'Running') { Start-Sleep -s 5 }",
      "& $env:SystemRoot\\System32\\Sysprep\\Sysprep.exe /oobe /generalize /mode:vm",
      "while($true) { $imageState = Get-ItemProperty HKLM:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Setup\\State | Select ImageState; if($imageState.ImageState -ne 'IMAGE_STATE_GENERALIZE_RESEAL_TO_OOBE') { Write-Output $imageState.ImageState; Start-Sleep -s 10  } else { break } }"
    ]
  }
]
}

**And this is the output im getting on my azdo pipeline**

<img width="763" alt="11" src="https://github.com/ScoopInstaller/Scoop/assets/136502493/77d9c285-9082-40de-a6f1-e987d09c61f5">
<img width="614" alt="22" src="https://github.com/ScoopInstaller/Scoop/assets/136502493/7bafcf21-97a8-4351-b167-589f650b3ce3">
<img width="612" alt="33" src="https://github.com/ScoopInstaller/Scoop/assets/136502493/3b6c5716-8242-44f5-9bc0-19d9cd26116d">

**Reply by HUMORCE** [reply]
Sorry, what is not being executed correctly?

What I see:
Scoop and program you specified is also installed correctly.

**Reply by ghost** [reply]
That is the problem, it shows that scoop is installed correctly but when i run the image through a Virtual machine, none of the tools are in there. I do not see scoop, or rancher desktop

**Reply by HUMORCE** [reply]
Try print username in Packer build, as well as in VM.

**Reply by ghost** [reply]
Could you elaborate please? I am still learning to work with packer so not sure how to print username.

**Reply by HUMORCE** [reply]
`$env:USERNAME`
